source/loader/layers/sanitizer/asan_interceptor.hpp

```diff
@@ -0,0 +1,163 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+ * See LICENSE.TXT
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * @file ur_sanitizer_layer.cpp
+ *
+ */
+
+#pragma once
+
+#include "common.hpp"
+#include "device_sanitizer_report.hpp"
+
+#include <map>
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+namespace ur_sanitizer_layer {
+
+enum USMMemoryType { DEVICE, SHARE, HOST, MEM_BUFFER };
```

Comment:
suggest to add unified prefix like UMT_ for better understanding and reduce confusion.

Reply:
I've comment this on another PR.